app/components/concerns/katalyst/tables/orderable.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module Katalyst
+  module Tables
+    # Adds drag and drop ordering to a table.
+    # See [documentation](/docs/orderable.md) for more details.
+    module Orderable
+      extend ActiveSupport::Concern
+
+      FORM_CONTROLLER = "tables--orderable--form"
+      ITEM_CONTROLLER = "tables--orderable--item"
+      LIST_CONTROLLER = "tables--orderable--list"
+
+      using HasHtmlAttributes
+
+      def self.extended(table)
+        table.class.config_component :orderable, default: FormComponent
+      end
+
+      def orderable?
+        @orderable.present?
+      end
+
+      def orderable
+        @orderable
+      end
+
+      def with_orderable(url:,
+                         id: "#{self.id}-orderable-form",
+                         scope: "order[#{collection.items.model_name.plural}]")
+        header_row_component.include(HeaderRow)
+        body_row_component.include(BodyRow)
+        @orderable = FormComponent.new(table: self, url: url, id: id, scope: scope)
```

Comment:
(opt) if we return `self` here, could we make this chainable? (Is that something we want to support?)
```ruby
table = TableComponent.new(collection:).with_orderable(url:)
```

Reply:
Good idea